src/main/scala/viper/silver/plugin/standard/inline/InlinePredicatePlugin.scala

```diff
@@ -0,0 +1,42 @@
+package viper.silver.plugin.standard.inline
+
+import viper.silver.ast.{Exp, LocalVar, Method, Node, PredicateAccess, Program}
+import viper.silver.plugin.{ParserPluginTemplate, SilverPlugin}
+
+class InlinePredicatePlugin extends SilverPlugin with ParserPluginTemplate {
+
+  private[this] val InlinePredicateKeyword = "inline"
+
+  override def beforeVerify(input: Program): Program = {
+    val declaredMethods = input.methods
+    val methodsWithExpandedPres = declaredMethods.map(expandPreconditions(_, input))
+
+    // TODO: actually rewrite program with methods with expanded predicates
+    input
+  }
+
+  private[this] def expandPreconditions(method: Method, program: Program): Method = {
```

Comment:
This is the method in which we take a method with predicates that are _not_ inlined, and returns a method with inlined preconditions.